pkg/router/types.go

```diff
@@ -50,7 +50,7 @@ var (
 )
 
 type headerFormatter interface {
-	format(requestInfo api.RequestInfo) string
+	format(headers types.HeaderMap, requestInfo api.RequestInfo) string
```

Comment:
I think `headers` and `requestInfo` are not needed,  just `ctx` to support variable..

Reply:
It seems that we didn't set the `requestInfo` into the `ctx`. And we set the request headers and response headers with different context key, but in the `format` method, we can't determine which header we are handing now,  so I think the parameters are necessary here.

Reply:
It seems that with `variable` api, we can get any value with `ctx`, I will try later!